src/main/java/uk/gov/gchq/magmacore/demo/JenaDatabaseDemo.java

```diff
@@ -52,32 +38,17 @@ public final class JenaDatabaseDemo {
      */
     public void run() {
         // Instantiate new in-memory Jena database.
-        final MagmaCoreJenaDatabase jenaDatabase = new MagmaCoreJenaDatabase();
-        jenaDatabase.register(HQDM);
-        jenaDatabase.register(RDFS);
-        jenaDatabase.register(REF_BASE);
-        jenaDatabase.register(USER_BASE);
-
-        // Create set of example data objects.
-        final List<Thing> objects = ExampleDataObjects.createDataObjects();
+        final var mcService = MagmaCoreServiceFactory.createWithJenaDatabase();
 
         // Add example data objects to dataset.
-        jenaDatabase.begin();
-        objects.forEach(object -> jenaDatabase.create(object));
-        jenaDatabase.commit();
+        ExampleDataObjects.populateExampleData(mcService);
 
         // Query database to check its populated.
-        jenaDatabase.begin();
-        final List<Thing> queryResults =
-                jenaDatabase.findByPredicateIriAndStringValue(ENTITY_NAME, "PersonB1_Bob");
-        jenaDatabase.abort();
+        final Thing queryResults = mcService.findByEntityName("PersonB1_Bob");
```

Comment:
Executing this demo produces the following error: 
> org.apache.jena.dboe.transaction.txn.TransactionException: Not in a transaction

Reply:
Fixed to run in a transaction.